source/core/index.ts

```diff
@@ -279,10 +277,7 @@ export interface Response<T = unknown> extends PlainResponse {
 }
 
 export interface RequestEvents<T> {
-	on(name: 'request', listener: (request: http.ClientRequest) => void): T;
-	on<R extends Response>(name: 'response', listener: (response: R) => void): T;
-	on<R extends Response, N extends NormalizedOptions>(name: 'redirect', listener: (response: R, nextOptions: N) => void): T;
-	on(name: 'uploadProgress' | 'downloadProgress', listener: (progress: Progress) => void): T;
+	on: ((name: 'request', listener: (request: http.ClientRequest) => void) => T) & (<R extends Response>(name: 'response', listener: (response: R) => void) => T) & (<R extends Response, N extends NormalizedOptions>(name: 'redirect', listener: (response: R, nextOptions: N) => void) => T) & ((name: 'uploadProgress' | 'downloadProgress', listener: (progress: Progress) => void) => T);
```

Comment:
This looks ugly.

Reply:
I think we can move the `&` to a separate line.

Reply:
Done: https://github.com/sindresorhus/got/pull/1345/commits/4dfe3e808558251ada2b0be10b2767c2098ad310